### PR TITLE
Execute spectral solver from the regular grid, instead of the dual grid

### PR DESCRIPTION
**Pull request by RemiLehe**
In the case `psatd.hybrid_mpi_decomposition=0`, instead of executing the spectral solver from the dual grid (`Efield_fp_fft`, `Bfield_fp_fft`), this PR executes the spectral solver directly from the regular grid (`Efield_fp`, `Bfield_fp`, etc.)

As a consequence the dual grid is not allocated anymore in this case, but the guard cells of the regular grid are increased (to the same size as the guard cells of the dual grid, so as to hold the extent of the stencil for the spectral solver).

Performance benefits:
- Use less memory, since we do not need to allocate the dual grid anymore.
- Avoid parallel copies back and forth between the dual grid and regular grid.

Note that I also added an automated test that checks the PSATD solver in the case `warpx.do_nodal=1`

### CODE REVIEW

File: Source/FieldSolver/WarpXFFT.cpp [hunk 1]
```diff
@@ -126,19 +126,6 @@ WarpX::AllocLevelDataFFT (int lev)
     FFTDomainDecomposition(lev, ba_fp_fft, dm_fp_fft, ba_valid_fp_fft[lev], domain_fp_fft[lev],
                            geom[lev].Domain());
```
Comment (RemiLehe) [inline Source/FieldSolver/WarpXFFT.cpp]: The code below has been moved to `WarpX.cpp`: `WarpX::AllocLevelMFs`

File: Source/WarpX.cpp [hunk 2]
```diff
@@ -570,8 +570,10 @@ WarpX::MakeNewLevelFromScratch (int lev, Real time, const BoxArray& new_grids,
     InitLevelData(lev, time);
 
 #ifdef WARPX_USE_PSATD
-    AllocLevelDataFFT(lev);
-    InitLevelDataFFT(lev, time);
```
Comment (RemiLehe) [inline Source/WarpX.cpp]: The code below allocates/initializes the dual grid.

**Comment by WeiqunZhang** [comment]
I am not convinced this is going to be a win.  There are consequences to having lots of ghost cells in `_fp`, `_cp` and `_aux`.  Do you have any proof this new way is better?  Could you run some tests?

**Comment by RemiLehe** [comment]
OK, good point.

Note that, from the point of view of *memory consumption*, I think this will be a win, since `Efield_fft_fp`, `Efield_fft_cp` already have that number of guard cells (i.e. `n_order_fft/2`) but also *duplicate* the valid cells. (Unless we do not allocate them, as is done in this PR)

However, from the point of view of time per iteration, I agree that it is unclear whether this is a win. I'll run performance tests, as you suggested.

**Comment by RemiLehe** [comment]
@RevathiJambunathan checked that this PR improves the performance significantly on GPU, in particular because it does not copy the data from the regular grid to the hybrid grid.

**Pull request merged**